src/datasource/gateway.js

```diff
@@ -0,0 +1,243 @@
+import Axios from "axios";
+import Errors from "./errors";
+
+/**
+ * Http Adapter.
+ * Configure your headers and what not in here.
+ * Interceptors and Network Layer middlewares should come in here.
+ */
+class AxiosProvider {
+  constructor(baseURL, config) {
+    this.http = Axios.create({
+      baseURL,
+      timeout: config.timeout || 100000,
+      headers: config.headers,
+    });
+  }
+}
+
+/**
+ * Http Control Layer.
+ * Throw your http control logic in here, i.e: if error, if data, if 201.
+ */
+class PocketQueriesController {
+  provider = null;
+
+  requests = {
+    getBalance: (address, height) => ({
+      url: '/v1/query/balance',
+      method: 'post',
+      data: {
+        address,
+        height,
+      }
+    }),
+    getTransaction: (id) => ({
+      url: '/v1/query/tx',
+      method: 'post',
+      data: {
+        hash: id,
+      }
+    }),
+    getApp: (address, height) => ({
+      url: '/v1/query/app',
+      method: 'post',
+      data: {
+        address,
+        height,
+      }
+    }),
+    getNode: (address, height) => ({
+      url: '/v1/query/node',
+      method: 'post',
+      data: {
+        address,
+        height,
+      }
+    }),
+    getAccountTxs: (address, received, prove, page, per_page) => ({
+      url: '/v1/query/accounttxs',
+      method: 'post',
+      data: {
+        address,
+        prove,
+        received,
+        page,
+        per_page,
+      }
+    }),
+    sendRawTx: (fromAddress, tx) => ({
+      url: '/v1/client/rawtx',
+      method: 'post',
+      data: {
+        address: fromAddress,
+        raw_hex_bytes: tx,
+      }
+    })
+  }
+
+  use(provider) {
+    this.provider = provider;
+    return this;
+  }
+
+  // looks ugly with the ifs
+  // but the gateway constantly responds with 200
+  // and responds with errors in response.data
+  // in a non-consistent form.
```

Comment:
Tracking this as an [issue](https://github.com/pokt-foundation/portal-api/issues/111)

Reply:
Not an issue, see issue